packages/server/integrations/jiraServer/JiraServerRestManager.ts

```diff
@@ -46,7 +72,34 @@ export default class JiraServerOAuth1Manager {
     return response
   }
 
-  async getProjects() {
-    return this.request('GET', '/rest/api/latest/project')
+  async parseJsonResponse(response: Response) {
+    const contentType = response.headers.get('content-type') || ''
+    if (!contentType.includes('application/json')) {
+      return new Error('Received non-JSON Jira Server Response')
+    }
+    const json = await response.json()
+    if (response.status !== 200) {
+      return new Error(
+        `Fetching projects failed with status ${response.status}, ${this.readError(json)}`
+      )
+    }
+
+    return json
+  }
+
+  async getProjects(): Promise<JiraServerRestProject[] | Error> {
+    const response = await this.request('GET', '/rest/api/latest/project')
+    const projects = this.parseJsonResponse(response)
```

Comment:
+1 doesn't change anything, but may be nice to `await` this or call it `projectsPromise`